lib/server.js

```diff
@@ -67,6 +67,10 @@ function CoAPServer(options, listener) {
     this._options.piggybackReplyMs = parameters.piggybackReplyMs
   }
 
+  if (!this._options.block2CacheExpiryMs || !isNumeric(this._options.block2CacheExpiryMs)) {
```

Comment:
I took the `piggybackReplyMs` part here and add `block2CheckExpiryMs` with same levels of checks. Maybe both should be better checked for validity. I'm not sure if is beneficial to even have this configurable, but just use `exchangeLifetime` unconditionally.

If value is too short (less than round-trip time), cached payloads will expiry and be removed before next block2 message is received. If unnessary large, cached payload stays in memory for longer than it should.

`exchangeLifetime` seems good default as that way cache entry is kept for the same length of time as maximum time for ACK for confirmable message. When client experiences connection problems, client could be in retransmit loop for next block2 request and with `exchangeLifetime` cache entry is stored reasonably long time.